client/components/AdoptionForm.js

```diff
@@ -0,0 +1,111 @@
+import React, { useState } from "react"
+
+const AdoptionForm = () => {
+  const [adoptionData, setAdoptionData] = useState({
+    name: "",
+    phoneNumber: "",
+    email: "",
+    homeStatus: "own"
+  })
+  let errorContainer = null
+  const [errors, setErrors] = useState([])
+  if (errors.length !== 0) {
+    errorContainer = errors.map(error => {
+      return <p>{error}</p>
+    })
+  }
+
+  const handleChange = (event) => {
+    setAdoptionData({
+      ...adoptionData,
+      [event.currentTarget.name]: event.currentTarget.value
+    })
+  }
+
+  const checkForErrors = () => {
+    const errorsArray = []
+    const fieldsData = Object.keys(adoptionData)
+    fieldsData.forEach(fieldName => {
+      if (adoptionData[fieldName] === "") {
+        errorsArray.push(fieldName)
+      }
+    })
+    return errorsArray
+  }
+
+  const handleSubmit = async (event) => {
+    event.preventDefault()
+    let errorsArray = checkForErrors()
+    if (errorsArray.length === 0) {
+      await addInfo()
+      setAdoptionData({
+        name: "",
+        phoneNumber: "",
+        email: "",
+        homeStatus: "own"
+      })
+    }
+    else {
+      setErrors(errorsArray)
+    }
+  }
+
+  const addInfo = async () => {
+    try {
+      const response = await fetch("/api/v1/petTypes", {
+        method: "POST",
+        headers: new Headers({
+          "Content-Type": "application/json"
+        }),
+        body: JSON.stringify(adoptionData)
+      })
+      if (!response.ok) {
+        if (response.status === 422) {
+          const body = await response.json()
+          return setErrors(body.errors)
+        } else {
+          const errorMessage = `${response.status} (${response.statusText})`
+          const error = new Error(errorMessage)
+          throw (error)
+        }
+      } else {
+        const body = await response.json()
+      }
+    } catch (err) {
+      console.error(err)
+    }
+  }
+
+  return (
+    <form onSubmit={handleSubmit}>
+
+      <div>{errorContainer}</div>
+
+      <label htmlFor="name">
+        Name:
+      <input id="name" name="name" value={adoptionData.name} onChange={handleChange} />
```

Comment:
the default `type` for an input is "text", which I imagine we're trying to use here, I would recommend explicitly setting the `type="text"` for these inputs for this form

we can also list each attribute on a separate line to help with readability